include/common/error_log.hpp

```diff
@@ -144,5 +153,96 @@ class FFDCFile
 
 }; // end of FFDCFile class
 
+/**
+ * @class FFDCFiles
+ *
+ * @brief This class is used to manage the created ffdc files
+ */
+class FFDCFiles
+{
+  public:
+    FFDCFiles() = delete;
+    FFDCFiles(const FFDCFiles&) = delete;
+    FFDCFiles& operator=(const FFDCFiles&) = delete;
+    FFDCFiles(FFDCFiles&&) = delete;
+    FFDCFiles& operator=(FFDCFiles&&) = delete;
+
+    /**
+     * @brief Used to created the FFDC files based on the given inputs
+     *
+     * @param[in] collectTraces - Used to indicate whether wants to collect
+     *                            the traces or not. If yes, create
+     *                            a FFDCFile for the same.
+     * @param[in] calloutsDetails - Used to create the FFDC file for
+     *                              the callouts if exist.
+     */
+    explicit FFDCFiles(const bool collectTraces, const json& calloutsDetails);
+
+    /**
+     * @brief Used to get the created FFDC files details to add in the error
+     *        log
+     *
+     * @param[out] ffdcFilesInfo - Used to return FFDCFiles with its
+     *                             information
+     *
+     * @return void
+     */
+    void transformFFDCFiles(FFDCFilesInfo& ffdcFilesInfo);
+
+  private:
+    /**
+     * @brief Used to store the list of created FFDCFile
+     */
+    std::vector<std::unique_ptr<FFDCFile>> _ffdcFiles;
+
+    /**
+     * @brief Helper API to get the given field value from
+     *        the systemd journal context
+     *
+     * @param[in] journal - The systemd journal to get the field value
+     * @param[in] fieldName - The field name to get its value
+     *
+     * @return The field value on success
+     *         Empty optional on failures
+     */
+    std::optional<std::string>
+        sdjGetTraceFieldValue(sd_journal* journal,
+                              const std::string& fieldName);
+
+    /**
+     * @brief Helper API to get the traces from systemd journal entries
+     *        based on the given field name and value.
+     *
+     * @param[in] fieldName - The field name to search.
+     * @param[in] fieldValue - The field value to search.
+     * @param[in] maxReqTraces - The maximum number of traces to fetch.
+     *
+     * @return The list of traces from the systemd journal on success
+     *         Emptry optional on failure
+     */
+    std::optional<std::vector<std::string>>
+        sdjGetTraces(const std::string& fieldName,
+                     const std::string& fieldValue,
+                     const unsigned int maxReqTraces = 10);
+
+    /**
+     * @brief Used to collect the systemd journal traces and create the
+     *        FFDCFile
+     *
+     * @return void
```

Comment:
is this functions throws exception during failure?  below code i can see this function called in try catch block.